backtrace-library/src/main/cpp/backtrace-native.cpp

```diff
@@ -48,6 +91,167 @@ JNIEXPORT jint JNI_OnLoad(JavaVM *jvm, void *reserved) {
 
 namespace /* anonymous */
 {
+    static void *
+    buffer_reader(int fd, int flags)
+    {
+        void *r;
+
+        r = mmap(NULL, BUFFER_SIZE, flags, MAP_SHARED, fd, 0);
+        if (r == MAP_FAILED)
+            abort();
+
+        return r;
+    }
+
+    static void *
+    buffer_create(void)
+    {
+        void *r;
+        int fd;
+
+        fd = bun_memfd_create("_backtrace_buffer", MFD_CLOEXEC);
+        if (fd == -1)
+            abort();
+
+        if (ftruncate(fd, BUFFER_SIZE) == -1)
+            abort();
+
+        buffer_fd = fd;
+
+        r = buffer_reader(buffer_fd, PROT_READ|PROT_WRITE);
+
+        return r;
+    }
+
+    static int
+    request_handler(pid_t tid)
+    {
+        // TODO revisit this logic
+        bool bun_initialized = bun_handle_init(&handle, BUN_BACKEND_LIBUNWINDSTACK);
+        if (!bun_initialized) {
+            return -1;
+        }
+
+        bun_buffer buf = { buffer_child, BUFFER_SIZE };
+
+        auto written = bun_unwind_remote(&handle, &buf, tid);
+
+        (void) written;
+
+        return -1;
```

Comment:
Are you sure request_handler should always (no matter what) return -1?

Reply:
That's a good point, what should we return on success for the bcd request_handler @KrzaQ ?

Reply:
@KrzaQ it's probably worth adding a comment to document this, I will do so now. But anyway the behavior is correct, thank you!